controllers/mhc/mhc_checker.go

```diff
@@ -50,30 +51,41 @@ const (
 )
 
 type checker struct {
-	client    client.Client
-	logger    logr.Logger
-	mhcStatus mhcStatus
+	client     client.Client
+	logger     logr.Logger
+	mhcStatus  mhcStatus
+	mhcRunning bool
 }
 
 var _ Checker = &checker{}
 
-// NeedDisableNHC checks if NHC needs to be disabled, because custom MHCs are configured in the cluster,
-// in order to avoid conflicts
-func (c *checker) NeedDisableNHC() (bool, error) {
+// Start will start the component and update the initial status
+func (c *checker) Start(ctx context.Context) error {
+	if err := c.UpdateStatus(); err != nil {
```

Comment:
IIUC this bit of code is called once (when Start is invoked).
Do we not support definition of MHC during runtime ?